pkg/integration/tests/branch/checkout_by_name.go

```diff
@@ -0,0 +1,38 @@
+package branch
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var CheckoutByName = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Try to checkout branch by name.",
+	ExtraCmdArgs: "",
+	Skip:         false,
+	SetupConfig:  func(config *config.AppConfig) {},
+	SetupRepo: func(shell *Shell) {
+		shell.
+			CreateNCommits(3).
+			NewBranch("@").
+			Checkout("master").
+			EmptyCommit("blah")
+	},
+	Run: func(shell *Shell, input *Input, assert *Assert, keys config.KeybindingConfig) {
+		input.SwitchToBranchesWindow()
+		assert.CurrentViewName("localBranches")
+
+		assert.MatchSelectedLine(Contains("master"))
+		input.NextItem()
+		assert.MatchSelectedLine(Contains("@"))
+		input.PressKeys(keys.Branches.CheckoutBranchByName)
+		assert.InPrompt()
+		assert.MatchCurrentViewTitle(Equals("Branch name:"))
+		input.Type("new-branch")
+		input.Confirm()
+		assert.InAlert()
```

Comment:
Could we assert on what the content of this alert is? That way it's easier to read the integration test and understand what's happening without needing to run the test yourself